h2/src/main/org/h2/mvstore/ChunkParser.java

```diff
@@ -0,0 +1,270 @@
+        	if (f != Field.ID) {
+        		Number value = mapChunkValues.get(f);
+    			setField(c, f, value != null ? value : getDefault(f, mapChunkValues));
+        	}
+        }
+        
+        return c;
+    }
+    
+    // Deserialize a Chunk to a map
+    private static Map<Field, Number> parseMap(String source) {
+    	final Map<Field, Number> mapChunkValues = new HashMap<>();
+    	final char[] value = new char[STRING_LEN_MAX_VALUE];
+
+        for (int i = 0, size = source.length(); i < size;) {
+            int startKey = i;
+            i = source.indexOf(':', i);
+            if (i < 0) {
+                throw DataUtils.newIllegalStateException(
+                        DataUtils.ERROR_FILE_CORRUPT, "Not a map: {0}", source);
+            }
+            
+            int valuePos = 0;
+            Field field = Field.getField(source, startKey, i - startKey);
+            i++;
+            while (i < size) {
+                char c = source.charAt(i++);
+                if (c == ',') {
+                    break;
+                } else if (c == '\"') {
+                    while (i < size) {
+                        c = source.charAt(i++);
+                        if (c == '\\') {
+                            if (i == size) {
+                                throw DataUtils.newIllegalStateException(
+                                        DataUtils.ERROR_FILE_CORRUPT,
+                                        "Not a map: {0}", source);
+                            }
+                            c = source.charAt(i++);
+                        } else if (c == '\"') {
+                            break;
+                        }
+
+                        // We're ignoring unrecognized keys
+                        if (field != null) {
+		                    value[valuePos++] = c;
+		                    
+		                    if (valuePos == STRING_LEN_MAX_VALUE) {
+		                        throw DataUtils.newIllegalStateException(
+		                                DataUtils.ERROR_FILE_CORRUPT,
+		                                "Value for key {0} is too long in {1}", field.key, source);
+		                    }
+                        }
+                    }
+                } else if (field != null) { // We're ignoring unrecognized keys
+                	value[valuePos++] = c;
+                    
+                    if (valuePos == STRING_LEN_MAX_VALUE) {
+                        throw DataUtils.newIllegalStateException(
+                                DataUtils.ERROR_FILE_CORRUPT,
+                                "Value for key {0} is too long in {1}", field.key, source);
+                    }
+                }
+            }
+
+            if (field != null) {
+	            // Parse the hex string into a Number
+	            
+	            // TODO With Java 8 use DataUtils.parse* method references
+	            // instead of the 'valueType'.
+	            if (Long.class.isAssignableFrom(field.valueType)) {
```

Comment:
might as well just use 
    Long.class ==  field.valueType
since we really only have two kinds of field